blocks/lists.js

```diff
@@ -141,13 +142,31 @@ Blockly.Blocks['lists_create_with'] = {
   },
   /**
    * Parse XML to restore the list inputs.
+   * Backwards compatible serialization implementation.
    * @param {!Element} xmlElement XML storage element.
    * @this {Blockly.Block}
    */
   domToMutation: function(xmlElement) {
     this.itemCount_ = parseInt(xmlElement.getAttribute('items'), 10);
     this.updateShape_();
   },
+  /**
+   * Returns the state of this block as a JSON serializable object.
+   * @return {{itemCount: number}} The state of this block, ie the item count.
+   */
+  saveExtraState: function() {
+    return {
+      'itemCount': this.itemCount_,
+    };
+  },
+  /**
+   * Applies the given state to this block.
+   * @param {*} state The state to apply to this block, ie the item count.
+   */
+  loadExtraState: function(state) {
+    this.itemCount_ = state['itemCount'];
+    this.updateShape_();
```

Comment:
@BeksOmega : any particular reason you call `updateShape_` here instead of `rebuildShape_`, as `domToMutation` does?

(See [this comment reopening issue #2037](https://github.com/google/blockly/issues/2037#issuecomment-2209637735) for why I am curious.)

Reply:
Best guess is that I didn't understand what `rebuildShape_` was trying to work around, and I figured `updateShape_` was more efficient. I don't think there's any behavioral reason to prefer `updateShape_`